src/directives/offense/controllerAttack.ts

```diff
@@ -0,0 +1,81 @@
+import {Directive} from '../Directive';
+import {profile} from '../../profiler/decorator';
+import {ScoutOverlord} from '../../overlords/core/scout';
+import {ControllerAttackerOverlord} from '../../overlords/offense/controllerAttacker';
+import {log} from '../../console/log';
+
+interface DirectiveControllerAttackMemory extends FlagMemory {
+	attackPos: protoPos[];
+	roomName: string;
+	readyAtTick: number;
+}
+
+
+@profile
+export class DirectiveControllerAttack extends Directive {
+
+	static directiveName = 'controllerAttack';
+	static color = COLOR_RED;
+	static secondaryColor = COLOR_YELLOW;
+
+	memory: DirectiveControllerAttackMemory;
+
+	constructor(flag: Flag) {
+		super(flag);
+
+		if (this.room) {
+			if (!this.memory.roomName) {
+				this.memory.roomName = this.room.name;
+			}
+		}
+
+		// TODO: Not have a scout at all times
+		this.overlords.scout = new ScoutOverlord(this);
+		this.overlords.controllerAttack = new ControllerAttackerOverlord(this);
+	}
+
+	get readyAtTick(): number {
```

Comment:
move this property to the overlord; directives shouldn't have much internal logic